pkg/operator/controller/externaldns/controller_test.go

```diff
@@ -128,12 +188,44 @@ func TestReconcile(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			cl := fake.NewClientBuilder().WithScheme(test.Scheme).WithRuntimeObjects(tc.existingObjects...).Build()
+			kubeFakeClient := kubernetesFakeClient.NewSimpleClientset()
+
 			r := &reconciler{
-				client: cl,
-				scheme: test.Scheme,
-				config: tc.inputConfig,
-				log:    zap.New(zap.UseDevMode(true)),
+				client:     cl,
+				scheme:     test.Scheme,
+				config:     tc.inputConfig,
+				log:        zap.New(zap.UseDevMode(true)),
+				kubeclient: kubeFakeClient.Discovery(),
```

Comment:
By the way, why do we need kubeclient in the reconciler? Isn't it again the leftover from the times when `IsOpenShift` was not global?